### PR TITLE
Pitch shooting

### PR DESCRIPTION
**Pull request by Josh-Westbrook**
## Justification
 We may have pitch by SA so it'd be good to make sure it'll work

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -87,6 +87,7 @@ public static class TurretConstants {
     public static final int PITCH_MOTOR_ID = 10;
     public static final int PITCH_ENCODER_ID = 1; // replace with actual ID
     public static final boolean IS_MOTOR_INVERTED = true;
+    public static final int defaultShotPitch = 40;
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/Constants.java:90]: UPPER CASE

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 2]
```diff
@@ -441,6 +441,14 @@ public void aimTurret() {
           tagDistanceEntry.log(distanceToTag);
         }
 
+        if (visionSubsystem.getTurretPitch(tagID) == 0) {
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:444]: Let's not call this twice, but instead call it once, and save the value into a local variable

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 3]
```diff
@@ -441,6 +441,14 @@ public void aimTurret() {
...
+          setPitchPos(Constants.TurretConstants.defaultShotPitch);
+          setTurretPos(0);
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:446]: Even though I imagine that we won't have a different default rotation than straight ahead, let's make this a constant as well.

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java (hunk 2 above)
Comment (caseyjbrotherton) [reply src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:444]: Can we add a comment on what does it mean to have getTurretPitch == 0?

It will help when we read this code later and try to understand what is happening.

**Review by caseyjbrotherton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 4]
```diff
@@ -71,24 +78,35 @@ public class TurretSubsystem extends SubsystemBase {
...
+      new DebugEntry<Double>(0.0, "Turret Goal Position (Radians)", this);
   private final DebugEntry<Double> turretVelocityEntry =
-      new DebugEntry<Double>(turretVelocity, "Turret Velocity", this);
+      new DebugEntry<Double>(turretVelocity, "Turret Velocity (RPM)", this);
   private final DebugEntry<Double> pitchPositionEntry =
-      new DebugEntry<Double>(pitchPosition, "Pitch Position", this);
+      new DebugEntry<Double>(pitchPosition, "Pitch Position (Rotations)", this);
   private final DebugEntry<Double> pitchGoalPositionEntry =
-      new DebugEntry<Double>(0.0, "Pitch Goal Position", this);
+      new DebugEntry<Double>(0.0, "Pitch Goal Position (Radians)", this);
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:99]: Now being logged in degrees

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 5]
```diff
@@ -71,24 +78,35 @@ public class TurretSubsystem extends SubsystemBase {
...
+      new DebugEntry<Double>(turretPosition, "Turret Position (Rotations)", this);
   private final DebugEntry<Double> turretGoalPositionEntry =
-      new DebugEntry<Double>(0.0, "Turret Goal Position", this);
+      new DebugEntry<Double>(0.0, "Turret Goal Position (Radians)", this);
   private final DebugEntry<Double> turretVelocityEntry =
-      new DebugEntry<Double>(turretVelocity, "Turret Velocity", this);
+      new DebugEntry<Double>(turretVelocity, "Turret Velocity (RPM)", this);
   private final DebugEntry<Double> pitchPositionEntry =
-      new DebugEntry<Double>(pitchPosition, "Pitch Position", this);
+      new DebugEntry<Double>(pitchPosition, "Pitch Position (Rotations)", this);
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:97]: Now being logged in degrees

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 6]
```diff
@@ -71,24 +78,35 @@ public class TurretSubsystem extends SubsystemBase {
...
+  private TunableNumber pitchTune;
+  private double pitchValue = 0;
+
   private final ShuffleboardTab turretTab = Shuffleboard.getTab(this.getName());
   private final DebugEntry<Double> turretPositionEntry =
-      new DebugEntry<Double>(turretPosition, "Turret Position", this);
+      new DebugEntry<Double>(turretPosition, "Turret Position (Rotations)", this);
   private final DebugEntry<Double> turretGoalPositionEntry =
-      new DebugEntry<Double>(0.0, "Turret Goal Position", this);
+      new DebugEntry<Double>(0.0, "Turret Goal Position (Radians)", this);
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:93]: Probably want to make these units match for testing/tuning

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 7]
```diff
@@ -119,6 +137,14 @@ public TurretSubsystem(RobotStateManager robotStateManager, VisionSubsystem visi
         CANSparkMax.SoftLimitDirection.kForward, (float) TurretConstants.PITCH_MAX_ANGLE_ROTATIONS);
     pitchMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, true);
     pitchMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kForward, true);
+    pitchMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus5, 20);
+    pitchAbsoluteEncoder = pitchMotor.getAbsoluteEncoder();
+    pitchAbsoluteEncoder.setPositionConversionFactor(1);
+    pitchAbsoluteEncoder.setInverted(false);
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:143]: Remove since we set inverted two lines down

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 8]
```diff
@@ -315,25 +352,25 @@ public void setTurretPos(double setpoint) {
...
                     Math.toRadians(Constants.TurretConstants.PITCH_MIN_ANGLE_DEGREES),
                     Math.toRadians(Constants.TurretConstants.PITCH_MAX_ANGLE_DEGREES)))
-            + pitchFeedForward.calculate(turretPosition, 0));
+     + pitchFeedForward.calculate(pitchPosition, 0));
   }
 
   public void holdPosition() {
     if (Constants.enabledSubsystems.turretRotationEnabled) setTurretPos(0);
-    if (Constants.enabledSubsystems.turretPitchEnabled) setPitchPos(0);
+    if (Constants.enabledSubsystems.turretPitchEnabled) setPitchPos(0.1);
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java:368]: Nitpick: Add code comment about magic number

File: src/main/java/frc/robot/RobotContainer.java [hunk 9]
```diff
@@ -304,7 +310,8 @@ private Command shootSpeaker() {
 
   private Command prepareToScoreSpeaker() {
     return Commands.parallel(
-        turretCommandFactory.getAimTurretCommand(), shooterCommandFactory.revShooter());
+        turretCommandFactory.moveUpwards(),
+        shooterCommandFactory.revShooter()); // TODO: just testing pitch from tunable number
```
Comment (azvanderpas) [inline src/main/java/frc/robot/RobotContainer.java:314]: This was changed just for testing. What do we want here for San Antonio?

**Review by azvanderpas — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java (hunk 9 above)
Comment (ModBoyex) [reply src/main/java/frc/robot/RobotContainer.java:314]: I think the idea is that we get lime light working and then have the appropriate command there, but idk

File: src/main/java/frc/robot/subsystems/shooterSubsystem/ShooterSubsystem.java [hunk 10]
```diff
@@ -303,7 +306,7 @@ public double getSpeedRightInRPM() {
           Constants.ShooterConstants.SHOOTER_IDLE_SPEED_RIGHT);
 
   public Trigger getBeamBreak() {
-    return new Trigger(beamBreak::get);
+    return new Trigger(() -> false);
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/shooterSubsystem/ShooterSubsystem.java:309]: Is the beam break working so we can revert this?
Auton needs this to shoot preload

File: src/main/java/frc/robot/Robot.java [hunk 11]
```diff
@@ -136,7 +136,7 @@ public void autonomousExit() {}
 
   @Override
   public void teleopInit() {
-    m_robotContainer.getDriveTrain().startVisionMeasures();
+    // m_robotContainer.getDriveTrain().startVisionMeasures();
```
Comment (azvanderpas) [inline src/main/java/frc/robot/Robot.java:139]: Should this be reverted?

File: src/main/java/frc/robot/Constants.java [hunk 12]
```diff
@@ -264,14 +265,14 @@ public static class enabledSubsystems {
...
+    public static final boolean usingPhoton = false;
     public static final boolean elvEnabled = true;
     public static final boolean signalEnabled = false;
-    public static final boolean shooterEnabled = false;
+    public static final boolean shooterEnabled = true;
     public static final boolean triggerEnabled = true;
     public static final boolean turretRotationEnabled = false;
     public static final boolean turretPitchEnabled = true;
-    public static final boolean climberEnabled = true;
+    public static final boolean climberEnabled = false;
```
Comment (azvanderpas) [inline src/main/java/frc/robot/Constants.java:275]: Don't forget to finalize these for comp

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java [hunk 13]
```diff
@@ -343,40 +350,30 @@ public void setTurretPos(double setpoint) {
...
+            Math.toRadians(Constants.TurretConstants.PITCH_MAX_ANGLE_DEGREES)));
     pitchGoalPositionEntry.log(Math.toDegrees(setpoint));
-    pitchMotor.setVoltage(
-        pitchPIDController.calculate(
-                pitchPosition,
-                MathUtil.clamp(
-                    setpoint,
-                    Math.toRadians(Constants.TurretConstants.PITCH_MIN_ANGLE_DEGREES),
-                    Math.toRadians(Constants.TurretConstants.PITCH_MAX_ANGLE_DEGREES)))
-            + pitchFeedForward.calculate(pitchPosition, 0));
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java]: Why was this changed?  I'm not understanding, we seem to have removed feed forward entirely?

Comment (azvanderpas) [reply src/main/java/frc/robot/subsystems/turretSubsystem/TurretSubsystem.java]: Its just setting the setpoint instead of doing both setpoint and new voltage calculation here. The setVoltage is moved down to periodic (with feedforward). Probably helps with trying to do a startEnd() Command instead of needing a run() Command.

File: src/main/java/frc/robot/Constants.java [hunk 14]
```diff
@@ -89,16 +89,19 @@ public static class TurretConstants {
...
     public static final double TURRET_KMAXOUTPUT = 1;
     public static final double TURRET_KMINOUTPUT = -1;
 
-    public static final int TURRET_MIN_ANGLE_DEGREES = 0;
-    public static final int TURRET_MAX_ANGLE_DEGREES = 40; // 56.2 - 11 measured
+    public static final int TURRET_MIN_ANGLE_DEGREES = -30;
+    public static final int TURRET_MAX_ANGLE_DEGREES = 30; // 56.2 - 11 measured
 
-    public static final double TURRET_CONVERSION_FACTOR = 0.25;
+    public static final double TURRET_CONVERSION_FACTOR = 0.18292682926829268292682926829268;
```
Comment (azvanderpas) [inline src/main/java/frc/robot/Constants.java:104]: If you want the most accuracy, instead of a long literal you should put the formula you used to get this number here and let the compiler do the math. Also somewhat self-documenting that way as well.

File: src/main/java/frc/robot/subsystems/trapElvSubsystem/TrapElvCommandFactory.java [hunk 15]
```diff
@@ -193,6 +193,17 @@ public BooleanSupplier getSourceBreak() {
...
 
+  public Command wristShooterRev() {
+    if (subsystem == null) return Commands.none();
+    return subsystem
+        .startEnd(
+            () -> {
+              subsystem.setWristState(TrapElvState.AMP_REV);
+            },
+            () -> {})
+        .withName("Pose AMP");
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/trapElvSubsystem/TrapElvCommandFactory.java:204]: update name & asProxy()?

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretCommandFactory.java [hunk 16]
```diff
@@ -104,20 +110,51 @@ public Command zeroTurretCommand() {
...
+        shortRangeShot(), longRangeShot(), () -> RSM.getRange() == RangeMode.SHORT);
+  }
+
+  public Command shortRangeShot() {
+    return subsystem.startEnd(
+        () -> {
+          if (Constants.enabledSubsystems.turretRotationEnabled)
+            subsystem.setTurretPos(Math.toRadians(00));
+          if (Constants.enabledSubsystems.turretPitchEnabled)
+            subsystem.setPitchPos(Math.toRadians(40));
```
[the remark this answers is not in the document]
Comment (azvanderpas) [reply src/main/java/frc/robot/subsystems/turretSubsystem/TurretCommandFactory.java:125]: And asProxy()?

File: src/main/java/frc/robot/subsystems/turretSubsystem/TurretCommandFactory.java [hunk 17]
```diff
@@ -104,20 +110,51 @@ public Command zeroTurretCommand() {
...
+        },
+        () -> {});
+  }
+
+  public Command longRangeShot() {
```
[the remark this answers is not in the document]
Comment (azvanderpas) [reply src/main/java/frc/robot/subsystems/turretSubsystem/TurretCommandFactory.java:130]: And asProxy()?

**Comment by caseyjbrotherton** [comment]
Functionality has been overwritten in the newest branches.  Closing pull request
